src/Main.elm

```diff
@@ -349,52 +322,37 @@ view model =
     }
 
 
-viewNavigation : Page -> Url.Url -> List (Html Msg)
-viewNavigation page url =
+viewNavigation : Route.Route -> List (Html Msg)
+viewNavigation route =
     let
-        preserveSearchOptions =
-            case page of
-                Packages model ->
-                    model.query
-                        |> Maybe.map (\q -> [ Url.Builder.string "query" q ])
-                        |> Maybe.withDefault []
-                        |> List.append [ Url.Builder.string "channel" model.channel ]
-
-                Options model ->
-                    model.query
-                        |> Maybe.map (\q -> [ Url.Builder.string "query" q ])
-                        |> Maybe.withDefault []
-                        |> List.append [ Url.Builder.string "channel" model.channel ]
```

Comment:
there is actually bug with my previous PR. I was not aware that there is another place where we build up URL. I think the decoding/encoding of search query gets broken when pages got switched. Sorry I haven't caught this before - it should have occurred to me that there must a be some mechanism for preserving query prams when changing the routes.